regression-test-all.sh

```diff
@@ -102,6 +103,7 @@ else
     : #echo '$CI'" = '$CI'; NOT doing continuous integration" >&2
 fi
 
+# this will probably need to be updated to reflect accurate information; CORES is supposed to be NUM_THREADS
```

Comment:
This is here because I've found that on some machines with 4 cores, it's OK to use all 4 when running make, but when running actual sampling using all 4 is too intense.

Reply:
I left the comment as a note to myself. What I call "NUM_THREADS" already has a 'word' for: CORES. So I use CORES throughout my regression tests instead of NUM_THREADS just for clarity